src/tss2-esys/esys_tr.c

```diff
@@ -369,6 +369,7 @@ Esys_TR_SetAuth(ESYS_CONTEXT * esys_context, ESYS_TR esys_handle,
     RSRC_NODE_T *esys_object;
     TSS2_RC r;
     _ESYS_ASSERT_NON_NULL(esys_context);
+    _ESYS_ASSERT_NON_NULL(authValue);
```

Comment:
I'm not 100% sure this is the right fix. IIRC, authValue NULL implies 0 length auth buffer.

Reply:
Commit message body lacking justification

Reply:
Right, just checked the spec and if auth is NULL the auth len should be set to zero. Patch updated.